client/scripts/controllers/chain/substrate/staking.ts

```diff
@@ -226,6 +257,93 @@ class SubstrateStaking implements StorageModule {
   public query(address: string): Observable<DeriveStakingQuery> {
     return this._Chain.query((api: ApiRx) => api.derive.staking.query(address));
   }
+  public get annualPercentRate(): Observable<ICommissionInfo> {
+    return this._Chain.api.pipe(
+      switchMap((api: ApiRx) => combineLatest(
+        of(api),
+        api.derive.staking.validators(),
+        api.query.staking.currentEra(),
+        from(this._app.chainEvents.rewards()),
+        api.derive.staking.electedInfo()
+      )),
+      flatMap((
+        [api, { validators: currentSet }, era, rewards, electedInfo]:
+        [ApiRx, DeriveStakingValidators, EraIndex, IReward, DeriveStakingElected]
+      ) => {
+        const commission: ICommissionInfo = {};
+
+        electedInfo.info.forEach(({ accountId, validatorPrefs }) => {
+          const key = accountId.toString();
+          commission[key] = (validatorPrefs.commission.unwrap() || new BN(0)).toNumber() / 10_000_000;
+          return commission[key];
+        });
+        // Different runtimes call for different access to stakers: old vs. new
+        const stakersCall = (api.query.staking.stakers)
+          ? api.query.staking.stakers
+          : api.query.staking.erasStakers;
+        // Different staking functions call for different function arguments: old vs. new
+        const stakersCallArgs = (account) => (api.query.staking.stakers)
+          ? account
+          : [era.toString(), account];
+        return combineLatest(
+          stakersCall.multi(currentSet.map((elt) => stakersCallArgs(elt.toString()))),
+          of(currentSet),
+          of(rewards),
+          of(commission)
+        );
+      }),
+      auditTime(100),
+      map(([exposures, accounts, rewards, commissions ] :
+        [Exposure[], AccountId[], IReward, ICommissionInfo ]) => {
+        // coins.div don't support fraction points. multiply and divide the same number will give fraction points.
+        const n = 100000;
+        const validatorRewards: ICommissionInfo = {};
+        let accountsTotalStake = new BN(0);
+
+        accounts.forEach((account, index) => {
+          const key = account.toString();
+          const exposure = exposures[index];
+          let stakeShare = 0;
+          let othersStake: BN = new BN(0);
+
+          exposure.others.forEach((indv) => {
```

Comment:
why not just use exposure.total? It seems you recalculate the total when it is already available?

Reply:
This gives the same result as those few lines:
```
exposure.own.toBn().muln(n).div(exposure.total.toBn()).toNumber() / n
```

Reply:
exposure.total is the total stake of a validator. And totalReward is the reward of validator from the reward's event.

Reply:
Yea, but you're just tallying up the stake shares it looks like `stakeShare` `otherStake` are the only parameters editted here.

Reply:
Yea, I am using `stakeShare` and `otherStake` for calculating the share of a validator in rewards. 
`stakeShare` = validator's share in percent from the staked amount
`otherStake` = nominators amount from the staked amount

adding more comments in source code for clarity.

Reply:
Why tally all the othersStake including one's own when you can just call `total`?

Reply:
Isn't exposure.own is the stake of validator? I am adding it to calculate the share of validators in stake amount left after deducting commission.